packages/web/src/scss/components/Modal/_ModalDialog.scss

```diff
@@ -46,14 +88,42 @@
         width: theme.$dialog-width-desktop;
     }
 
+    // @deprecated The "uniform" dialog variant is deprecated and will be removed in the next major release.
+    // Migration:
+    // 1. Remove the feature flag and make the uniform dialog the default.
+    // 2. Adjust transitions below.
+    @if feature-flags.$modal-enable-uniform-dialog {
+        @include -uniform-modal-dialog();
+    }
+
     @media (prefers-reduced-motion: no-preference) {
         transition-property: bottom, width, border-radius, transform; // 1.
         transition-duration: inherit;
     }
 }
 
-.ModalDialog--expandOnMobile {
-    @include breakpoint.down(map.get(theme.$breakpoints, tablet)) {
-        height: theme.$dialog-max-height;
+// @deprecated The "uniform" dialog variant is deprecated and will be removed in the next major release.
+// Migration: Remove the feature flag and make the uniform dialog the default.
+@if feature-flags.$modal-enable-uniform-dialog {
+    .ModalDialog--docked {
```

Comment:
I am not 100 % sure we shouldn't somehow mention in the modifier name that docked is the ModalDialog only on the mobile breakpoint.

Reply:
OK, let's go with `dockOnMobile` (similar to `expandOnMobile`).